File: hygiene_nerds_backend/src/main/java/com/universe/hygienenerds/hygiene_nerds_backend/dto/ProductDto.java
package com.universe.hygienenerds.hygiene_nerds_backend.dto;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@ToString
public class ProductDto {
    private int id;
    private String name;
    private String brandName;
    private String description;
    private double price;
    private String imageUrl;
    private String category;
    private int stock;
}

File: hygiene_nerds_frontend/src/types/product.dto.ts
export interface ProductDto {
  id: number;
  name: string;
  description: string;
  brandName: string;
  price: number;
  imageUrl: string;
  category: string;
  stock: number;
}

export interface SpecialPackage {
  id: number;
  packageName: string;
  duration: number;
  expirationDate: string;
  offeredProducts: ProductDto[];
  discount?: number; // Calculated field
  price?: number; // Calculated field
  originalPrice?: number; // Calculated field
}
